docs/order.rst

```diff
@@ -0,0 +1,142 @@
+.. _order:
+
+Ordering
+========
+
+Ordering is very similar to filtering. It is provided via method ``_order_queryset`` that you can override to change
+its default implementation. Resource class ``BaseModelResource`` again uses manager to simplify custom modifications::
+
+    class BaseModelResource(DefaultRESTModelResource, BaseObjectResource):
+        order_manager = DefaultModelOrderManager()
+
+        def _order_queryset(self, qs):
+            if self.order_manager:
+                return self.order_manager.order(self, qs, self.request)
+            else:
+                return qs
+
+Method ``_order_queryset`` uses manager to order/sort output data. You can change manager with ``order_manager``
+attribute. If you set ``order_manager`` to ``None``. Filtering will be disabled. Default order manager is
+``DefaultModelOrderManager``.
+
+Order manager
+-------------
+
+Order manager purposes are:
+
+* parse input data that contains information about ordering and split this data to pairs <order identifier, order direction>
+* check if concrete identifier is allowed to order
+* convert identifiers to django ordering identifiers
+* finally apply ordering on django QuerySet
+
+Allowed identifiers defines ``order_fields`` or ``extra_order_fields`` that can be defined on model or resource (or both).
+
+DefaultModelOrderManager
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is only one pre-implemented Pyston ordering manager. Input data is parsed from querystring named ``order`` or
+HTTP header named ``X-Order``. Both have same format of value. Order terms are split with comma char, each term is
+consits  of direction char and identifier. missing direction char means ascending, char '-' means descending direction.
+Identifiers are very similar to django identifiers where relations are joined with char '__'. Example::
+
+    first_name,-last_name,created_issues__created_at
+
+
+Ordering field configuration
+----------------------------
+
+By default fields that can be ordered is generated as join of ``order_fields`` and ``extra_order_fields``. But you can
```

Comment:
By default the fields that can be ordered are generated as a join of the ``order_fields`` and ``extra_order_fields``. But you can change this behaviour by overriding the method get_order_fields_rfs. The values of ``order_fields`` and ``extra_order_fields`` are first taken from the resource and if they are not set in the resource they are obtained from the model RESTMeta. If ``order_fields`` is not defined in RESTMeta, it is replaced with response of the method ``get_allowed_fields_rfs`` which returns all fields that the client of the resource can read.